contracts/src/v0.8/workflow/dev/WorkflowRegistryManager.sol

```
@@ -66,6 +67,12 @@ contract WorkflowRegistryManager is Ownable2StepMsgSender, ITypeAndVersion {
     string memory typeVer = _getTypeAndVersionForContract(contractAddress);
     uint32 latestVersionNumber = ++s_latestVersionNumber;
 
+    // Check if the contract is already registered. If it is, you can just activate that existing version.
+    bytes32 key = keccak256(abi.encodePacked(contractAddress, chainID));
+    if (s_versionNumberByAddressAndChainID[key] != 0) {
+      revert ContractAlreadyRegistered(contractAddress, chainID);
+    }
+
```

Nit: i would recommend doing this check in the beginning of the function, so we fail fast, fail early.

---

ah good call let me move it up!